### PR TITLE
Clear remote ref with auto merge

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### CODE REVIEW

File: vendor/github.com/bitrise-io/go-utils/command/git/commands.go [hunk 1]
```diff
@@ -162,3 +162,10 @@ func (g *Git) SparseCheckoutSet(opts ...string) *command.Model {
 	args = append(args, opts...)
 	return g.command(args...)
 }
+
```
Comment (ofalvai) [inline vendor/github.com/bitrise-io/go-utils/command/git/commands.go:165]: How did this update if there is no go.sum change? Is the vendor folder consistent?

File: gitclone/checkout_method_pr_auto_merge_branch.go [hunk 2]
```diff
@@ -40,8 +40,15 @@ func (c checkoutPRMergeRef) do(gitCmd git.Git, fetchOpts fetchOptions, fallback
 	// https://git-scm.com/book/en/v2/Git-Internals-The-Refspec
 	refSpec := fmt.Sprintf("%s:%s", c.remoteMergeRef(), c.localMergeRef())
 
+	// delete remote branch ref
```
Comment (ofalvai) [inline gitclone/checkout_method_pr_auto_merge_branch.go:43]: Can you add a comment explaining why we need this? It's not trivial and no one will remember in a year

File: gitclone/checkout_method_pr_auto_merge_branch.go [hunk 3]
```diff
@@ -40,8 +40,15 @@ func (c checkoutPRMergeRef) do(gitCmd git.Git, fetchOpts fetchOptions, fallback
...
+	// $ git update-ref -d refs/remotes/pull/7/merge
+	err := deleteRef(gitCmd, c.remoteMergeRef())
+	if err != nil {
+		return fmt.Errorf("failed to delte ref: %w", err)
```
Comment (ofalvai) [inline gitclone/checkout_method_pr_auto_merge_branch.go:47]: The doc says that it first verifies that the ref exists before deleting it.. Which means we can't just run it all the time, it's going to bubble up the error if there is nothing to clean up.

File: vendor/github.com/bitrise-io/go-utils/command/git/commands.go (hunk 1 above)
Comment (lpusok) [reply vendor/github.com/bitrise-io/go-utils/command/git/commands.go:176]: Updated with correct vendor updates.

File: gitclone/checkout_method_pr_auto_merge_branch.go (hunk 2 above)
Comment (lpusok) [reply gitclone/checkout_method_pr_auto_merge_branch.go]: Added comment.

File: gitclone/checkout_method_pr_auto_merge_branch.go (hunk 3 above)
Comment (lpusok) [reply gitclone/checkout_method_pr_auto_merge_branch.go]: Tested on macos/linux, the git command does not fail.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**